packages/std/src/errors.rs

```diff
@@ -124,14 +124,17 @@ mod test {
         };
     }
 
-    /// How can this work?
+    /// The deseralizer in from_slice can perform zero-copy deserializations (https://serde.rs/lifetimes.html).
```

Comment:
There is a `serde::de::DeserializeOwned` trait that states that your object hold no references to the passed in slice.

You may want to use this a trait bound (this is a specialization of `Deserialize`)

Reply:
Right. However, that would affect all usages of `from_slice`, not only the StdError. I think it is better to just make error fields owned consistently and keep `Deserialize` for other usecases